File: swift-edition/liner-swift-edition/liner-swift-edition/solids/platform.swift
//
//  File.swift
//  liner-swift-edition
//
//  Created by easontek2398 on 8/6/22.
//

import Foundation
import Raylib

class Platform {
    var rect: Rectangle? = nil
    init(x posx: Float, y posy: Float, width Width: Float, height Height: Float) {
        self.rect = Rectangle(x: posx, y: posy, width: Width, height: Height)
    }
    func Draw() {
        
    }
    func Collision() {
        
    }
    func CeilingCollision() {
        
    }
}

File: src/elements/platform.py
from raylib import *
from pyray import Rectangle

class Platform:
    def __init__(self, posx: int, posy: int, width: int, height: int, color: list[int]):
        self.rect = Rectangle(posx,posy,width,height)
        self.color = color
    
    def draw(self):
        DrawRectangleRec(self.rect, self.color)

    def collision(self, player) -> bool:
        if CheckCollisionRecs(self.rect, player):
            return True
        else:
            return False